packages/replayio/src/utils/installation/getLatestReleases.ts

```diff
@@ -5,24 +5,24 @@ import { runtimeMetadata } from "./config";
 import { debug } from "./debug";
 import { Release } from "./types";
 
-const { platform, runtime } = runtimeMetadata;
+const { platform, runtime, architecture } = runtimeMetadata;
 
 export async function getLatestRelease() {
   debug("Fetching release metadata");
 
   const response = await fetch(`${replayAppHost}/api/releases`);
   const json = (await response.json()) as Release[];
-  const latestRelease = json.find(release => {
-    if (release.platform === platform && release.runtime === runtime) {
-      if (platform === "macOS") {
-        return process.arch.startsWith("arm") === release.releaseFile.includes("arm");
-      } else {
-        return true;
-      }
-    }
-  });
+  const latestRelease = json.find(
+    release =>
+      release.platform === platform &&
+      release.runtime === runtime &&
+      (release.architecture === architecture || release.architecture === null)
+  );
 
-  assert(latestRelease, `No release found for ${platform}:${runtime}`);
+  debug("Latest release", latestRelease);
+  if (!latestRelease) {
+    throw new Error(`No release found for ${platform}:${runtime}`);
+  }
```

Comment:
`assert` still throws an error - so [this test case](https://github.com/replayio/replay-cli/pull/451/files#diff-ee1a1757b4a83febf95bb5c25c7780d53a29311a86aebde889dde4e5195af90aR23) passes even with an `assert`:
```suggestion
  assert(latestRelease, `No release found for ${platform}:${runtime}`);
```